llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -1983,7 +1988,11 @@ PassBuilder::buildLTODefaultPipeline(OptimizationLevel Level,
 
   // If we didn't decide to inline a function, check to see if we can
   // transform it to pass arguments by value instead of by reference.
-  MPM.addPass(createModuleToPostOrderCGSCCPassAdaptor(ArgumentPromotionPass()));
+  CGSCCPassManager CGPM;
+  CGPM.addPass(ArgumentPromotionPass());
+  CGPM.addPass(CoroSplitPass(Level != OptimizationLevel::O0));
+  CGPM.addPass(CoroAnnotationElidePass());
```

Comment:
should those passes be wrapped in a `CoroConditionalWrapper` pass?

@aeubanks I saw that you originally introduced `CoroConditionalWrapper` in 9bd66b312c3e0e539ac6c85c1bacd17bdc731614, but used it in `buildO0DefaultPipeline` only. Should this be wrapped around all coro-related passes, to avoid potential performance regressions?

Reply:
I think it should be out of scope of the PR.
BTW  what is the slowest of them? THouse in CGSCCPassManager related?
I think CoroEarlyPass CoroCleanupPass should be fast if nothing to do.

Also if I read it correctly  https://github.com/llvm/llvm-project/issues/54117, it was done to avoid GlobalDCEPass?

Reply:
compile-time-wise, it shouldn't matter. CoroConditionalWrapper was added to preventing having to construct the LazyCallGraph just to throw it away immediately, but that's not relevant here. AFAIK the coroutine passes all bail early if there aren't any coroutine intrinsics